src/atomate2/abinit/schemas/anaddb.py

```diff
@@ -0,0 +1,398 @@
+        Metadata tags for this task document
+    """
+
+    dir_name: Optional[str] = Field(
+        None, description="The directory for this Abinit task"
+    )
+    completed_at: Optional[str] = Field(
+        None, description="Timestamp for when this task was completed"
+    )
+    output: Optional[OutputDoc] = Field(
+        None, description="The output of the final calculation"
+    )
+    structure: Union[Structure] = Field(
+        None, description="Final output atoms from the task"
+    )
+    state: Optional[TaskState] = Field(None, description="State of this task")
+    event_report: Optional[events.EventReport] = Field(
+        None, description="Event report of this abinit job."
+    )
+    included_objects: Optional[list[AbinitObject]] = Field(
+        None, description="List of Abinit objects included with this task document"
+    )
+    abinit_objects: Optional[dict[AbinitObject, Any]] = Field(
+        None, description="Abinit objects associated with this task"
+    )
+    task_label: Optional[str] = Field(None, description="A description of the task")
+    tags: Optional[list[str]] = Field(
+        None, description="Metadata tags for this task document"
+    )
+
+    @classmethod
+    def from_directory(
+        cls,
+        dir_name: Path | str,
+        additional_fields: dict[str, Any] = None,
+        **abinit_calculation_kwargs,
+    ) -> AnaddbTaskDoc:
+        """Create a task document from a directory containing Abinit/anaddb files.
+
+        Parameters
+        ----------
+        dir_name: Path or str
+            The path to the folder containing the calculation outputs.
+        additional_fields: Dict[str, Any]
+            Dictionary of additional fields to add to output document.
+        **abinit_calculation_kwargs
+            Additional parsing options that will be passed to the
+            :obj:`.Calculation.from_abinit_files` function.
+
+        Returns
+        -------
+        .AnaddbTaskDoc
+            A task document for the calculation.
+        """
+        logger.info(f"Getting task doc in: {dir_name}")
+
+        if additional_fields is None:
+            additional_fields = {}
+
+        dir_name = Path(dir_name)
+        task_files = _find_abinit_files(dir_name)
+
+        if len(task_files) == 0:
+            raise FileNotFoundError("No Abinit files found!")
+
+        calcs_reversed = []
+        all_abinit_objects = []
+        for task_name, files in task_files.items():
+            calc_doc, abinit_objects = Calculation.from_abinit_files(
+                dir_name, task_name, **files, **abinit_calculation_kwargs
+            )
+            calcs_reversed.append(calc_doc)
+            all_abinit_objects.append(abinit_objects)
+
+        tags = additional_fields.get("tags")
+
+        dir_name = get_uri(dir_name)  # convert to full uri path
+        dir_name = strip_hostname(
+            dir_name
+        )  # VT: TODO to put here?necessary with laptop at least...
```

Comment:
When I remove these lines and try to do linked calculations, I get this:
```
File "/home/vtrinquet/Softwares_Packages/Github/atomate2/src/atomate2/abinit/files.py", line 118, in load_abinit_input
    raise NotImplementedError(
NotImplementedError: Cannot load AbinitInput from directory PC-SE22-351:/tmp/tmppwb7e16s/job_2024-07-31-14-50-15-596245-46771 without abinit_input.json file.
```

Reply:
I see the problem. However this is the output that will be stored in the DB and it makes sense that it will contain the hostname. Also in this way it will be consistent with the other codes.
Instead do the `strip_hostname` in `load_abinit_input` (and wherever it might be needed). Leave a #TODO there to specify that this could be handled in the future (could probably be done with the FileClient).
Change this also in the other schemas.

Reply:
done